KitePrintSDK/src/main/java/ly/kite/KiteSDK.java

```diff
@@ -368,9 +411,12 @@ static private void setParameter( Context context, Scope scope, String prefix, S
     {
     String key = getParameterKey( prefix, name );
 
+      //////// Encryption initialiser //////////
+      SecurePreferences pref = new SecurePreferences(ENCRYPTION_KEY);
+
     scope.sharedPreferences( context )
             .edit()
-            .putStringSet( key, stringSet )
+            .putStringSet( pref.encrypt(key), stringSet )
```

Comment:
You're not encrypting stringSet... As it's a HashSet of strings, iterate through and encrypt each string individually in the set. Then do the same when decrypting